internal/metrics/metrics.go

```diff
@@ -46,9 +58,48 @@ var (
 		[]string{dnsRecordNameLabel, dnsRecordNamespaceLabel, dnsRecordRootHost, dnsRecordDelegating})
 )
 
+func NewAuthoritativeRecordSpecInfoMetric(dnsRecord *v1alpha1.DNSRecord) (*authoritativeRecordSpecInfoMetric, error) {
+	if dnsRecord == nil {
+		return nil, errors.New("dnsRecord nil pointer")
+	}
+
+	metric := &authoritativeRecordSpecInfoMetric{
+		RootHost:  dnsRecord.Spec.RootHost,
+		Name:      dnsRecord.Name,
+		NameSpace: dnsRecord.Namespace,
+	}
+	err := metric.calculateSha(dnsRecord)
+	if err != nil {
+		return nil, err
+	}
+	return metric, nil
+}
+
+type authoritativeRecordSpecInfoMetric struct {
+	RootHost  string
+	Sha       string
+	Name      string
+	NameSpace string
+}
+
+func (m *authoritativeRecordSpecInfoMetric) calculateSha(dnsRecord *v1alpha1.DNSRecord) error {
+	if dnsRecord == nil {
+		return errors.New("dnsRecord nil pointer")
+	}
+
+	specString := hash.GetCanonicalString(dnsRecord.Spec)
+	m.Sha = hash.ToBase36HashLen(specString, 6)
```

Comment:
In terms of metrics does the length of the value have a significant impact on the expense of this metric? I'm wondering would you be better to use `hash.ToBase36Hash`, or some other hashing algorithm,  to decrease the possibility of two different strings returning the same value.

Reply:
OK, i don't really mind what we use. The shortened version of the UID was being used because of limitations in the txt registry. This limitation no longer exist, and we probably should be using the full length hash for those instead.